rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/user/UserController.java

```diff
@@ -207,4 +211,59 @@ public RepositoryLinksResource process(RepositoryLinksResource resource) {
     private HalResource<User> createHalUser(User sw360User) {
         return new HalResource<>(sw360User);
     }
+
+    @Operation(
+            summary = "List all of the service's users.",
+            description = "List all of the service's users.",
+            tags = {"Users"}
+    )
+    @GetMapping(value = USERS_URL + "/name")
+    public ResponseEntity<CollectionModel<EntityModel<User>>> getUsersByName(
+            Pageable pageable,
+            HttpServletRequest request,
+            @Parameter(description = "fullName of the users")
+            @RequestParam(value = "givenname", required = false) String givenname,
+            @RequestParam(value = "email", required = false) String email,
+            @Parameter(description = "luceneSearch parameter to filter the users.")
+            @RequestParam(value = "luceneSearch", required = false) boolean luceneSearch
+            ) throws TException, URISyntaxException, PaginationParameterException,  ResourceClassNotFoundException {
+        PaginationResult<User> paginationResult = null;
+        if (luceneSearch) {
+            Map<String, Set<String>> filterMap = new HashMap<>();
+            if (CommonUtils.isNotNullEmptyOrWhitespace(givenname)) {
+                Set<String> values = CommonUtils.splitToSet(givenname);
+                values = values.stream().map(LuceneAwareDatabaseConnector::prepareWildcardQuery)
+                        .collect(Collectors.toSet());
+                filterMap.put(User._Fields.GIVENNAME.getFieldName(), values);
+            }
+            if (CommonUtils.isNotNullEmptyOrWhitespace(email)) {
+                Set<String> values = CommonUtils.splitToSet(email);
+                values = values.stream().map(LuceneAwareDatabaseConnector::prepareWildcardQuery)
+                        .collect(Collectors.toSet());
+                filterMap.put(User._Fields.EMAIL    .getFieldName(), values);
+            }
+            List<User> userByGivenName = userService.refineSearch(filterMap);
+            paginationResult = restControllerHelper.createPaginationResult(request, pageable, userByGivenName, SW360Constants.TYPE_USER);
+        }else {
+            List<User> sw360Users = userService.getAllUsers();
+            paginationResult = restControllerHelper.createPaginationResult(request, pageable, sw360Users, SW360Constants.TYPE_USER);
+        }
+
+        List<EntityModel<User>> userResources = new ArrayList<>();
```

Comment:
Seems like if name is not present then it will return all the users instead it should query(not lucene) the database for the user.